snapcraft/snap_config.py

```diff
@@ -15,54 +15,30 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 """Snap config file definitions and helpers."""
-from typing import Any, Dict, Literal, Optional
+from typing import Literal, Optional
 
+import craft_application.models
 import pydantic
 from craft_cli import emit
 from snaphelpers import SnapConfigOptions, SnapCtlError
 
 from snapcraft.utils import is_snapcraft_running_from_snap
 
 
-class SnapConfig(pydantic.BaseModel, extra=pydantic.Extra.forbid):
+class SnapConfig(craft_application.models.CraftBaseModel):
     """Data stored in a snap config.
 
     :param provider: provider to use. Valid values are 'lxd' and 'multipass'.
     """
 
-    provider: Optional[Literal["lxd", "multipass"]] = None
+    provider: Literal["lxd", "multipass"] | None = None
```

Comment:
I'm not entirely sure this will work, but if it does...

```suggestion
    provider: Annotated[Literal["lxd", "multipass"], pydantic.StringConstraints(to_lower=True)] | None = None
```

you can get rid of the validator that does this.